charts/skypilot/values.yaml

```diff
@@ -102,12 +102,68 @@ ingress-nginx:
 
 rbac:
   # Configure the RBAC permissions for the SkyPilot API server
-  rules:
-    # Default configuration with full access - override for restricted environments
-    - apiGroups: ["*"]
-      resources: ["*"]
-      verbs: ["*"]
-
+  # TODO(aylei): Do not setup rbac policies when rbac.kubernetesCredentials.userKubeconfig=true
+  # Addtional grantedNamespaces other than the release namespace (where API server is deployed).
+  # Different namespaces may need different permissions, but it would be complex and error-prone to manage rules for each namespace.
+  # Instead, we aggregate all the permissions needed across namespaces and grant them to all the namespaces listed.
+  grantedNamespaces:
+  - namespace: "skypilot-system"
+  # Namespace-scoped rules that are granted to the release namespace and rbac.grantedNamespaces.
+  namespaceRules:
+    # Required for managing pods and their lifecycle
+    - apiGroups: [ "" ]
+      resources: [ "pods", "pods/status", "pods/exec", "pods/portforward" ]
+      verbs: [ "*" ]
+    # Required for managing services for SkyPilot Pods
+    - apiGroups: [ "" ]
+      resources: [ "services" ]
+      verbs: [ "get", "list", "watch", "create", "update", "patch", "delete" ]
+    # Required for managing SSH keys
+    - apiGroups: [ "" ]
+      resources: [ "secrets" ]  
+      verbs: [ "get", "list", "watch", "create", "update", "patch", "delete" ]
```

Comment:
Do we need create/update for secrets?

Reply:
Yes, for managing ssh key secrets [We create and patch ](https://github.com/skypilot-org/skypilot/blob/633e16611f2f858dc27c9eae2f410811e0bc714c/sky/authentication.py#L429)